### DISCUSSION TITLE
Assig_coords in xarray and data array object error

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#6508] [Q&A] [answered]
Hello! This is from an earlier discussion and I am trying to regroup a monthly data set (time, lat, and lon) into year, month, lat, and lon:
(https://github.com/pydata/xarray/discussions/5119)

Some really helpful code was provided in one of the responses to do this (answered by @mathause):

` def to_monthly(ds):
    year = ds.time.dt.year
    month = ds.time.dt.month`

    # assign new coords
    ds = ds.assign_coords(year=("time", year), month=("time", month))

    # reshape the array to (..., "month", "year")
    return ds.set_index(time=("year", "month")).unstack("time")  `

air = xr.tutorial.open_dataset("air_temperature").resample(time="M").mean() `

However when I try calling the function I am getting an error that says:

`to_monthly(air)`

`TypeError: Using a DataArray object to construct a variable is ambiguous, please extract the data using the .data property.`

which looks to be appearing at the `ds.assign_coords` part. I recently updated xarray to the latest version as well, and I still have the issue.
Any ideas on a workaround or where this might be from? Thank you so much!

**Comment by mathause** [comment, marked as answer]
Without testing - can you try:

```python
ds = ds.assign_coords(year=("time", year.data), month=("time", month.data))
```

If this works I should also update my other answer - so I'd appreciate feedback.

**Reply by pecos27** [reply]
Great, this works perfectly! Guess I should listen to what the error is saying more next time :D, thank you so much!
